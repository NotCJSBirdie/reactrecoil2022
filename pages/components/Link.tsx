import React from "react";
import { useState } from "react";

const Link = () => {
  const [status, setStatus] = useState("I am the default Link");

  const onMouseEnter = () => {
    setStatus("The mouse has entered!!");
  };

  const onMouseLeave = () => {
    setStatus("The mouse had left!!");
  };

  return (
    <div>
      <a onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
        {status}
      </a>
    </div>
  );
};

export default Link;
