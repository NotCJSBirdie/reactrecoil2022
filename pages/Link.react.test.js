import React from "react";
import renderer from "react-test-renderer";
import Link from "./components/Link";

test("Link changes the class when hovered", () => {
  const component = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>
  );

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
