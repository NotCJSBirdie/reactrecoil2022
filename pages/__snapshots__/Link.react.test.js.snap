// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div>
  <a
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    I am the default Link
  </a>
</div>
`;

exports[`Link changes the class when hovered 2`] = `
<div>
  <a
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    I am the default Link
  </a>
</div>
`;

exports[`Link changes the class when hovered 3`] = `
<div>
  <a
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    I am the default Link
  </a>
</div>
`;
